pytorch_lightning/utilities/auto_restart.py

```diff
@@ -76,10 +79,13 @@ def __iter__(self) -> Iterator[Any]:
 
                 # when the current index matching the current_iteration, we have "fast forwarded" the sampler.
                 if self._current_iteration <= i:
+                    if self._current_iteration == i:
+                        self.restarting = False
```

Comment:
This is considered and side effect. Can't we do it where this is called instead?

Reply:
Yes, feel free to change it.

Reply:
I don't have enough context to know where it should be done

Reply:
Moved up.